src/Configuration/src/Placeholder/PlaceholderResolverProvider.cs

```diff
@@ -77,13 +76,13 @@ public PlaceholderResolverProvider(IList<IConfigurationProvider> providers, ILog
     /// <returns>
     /// <c>true</c> if a value for the specified key was found, otherwise <c>false</c>.
     /// </returns>
-    public bool TryGet(string key, out string value)
+    public bool TryGet(string key, out string? value)
     {
-        ArgumentGuard.NotNull(key);
+        ArgumentException.ThrowIfNullOrEmpty(key);
         EnsureInitialized();
 
         string? originalValue = Configuration![key];
-        value = PropertyPlaceholderHelper.ResolvePlaceholders(originalValue, Configuration);
+        value = PropertyPlaceholderHelper.ResolvePlaceholders(originalValue, Configuration, NullLogger.Instance);
```

Comment:
Shouldn't we be passing a logger from the factory in here?

Reply:
Yes, we should, good catch. This is where it pays off to make the parameter non-optional.